nodejs/aws-infra/experimental/clusterService.ts

```diff
@@ -0,0 +1,337 @@
+        }
+    }
+
+    return loadBalancers;
+}
+
+function combineLoadBalancers(
+        e1: pulumi.Input<mod.LoadBalancers | undefined>,
+        e2: pulumi.Input<mod.LoadBalancers | undefined>): mod.LoadBalancers {
+    return pulumi.all([e1, e2]).apply(([e1, e2]) => {
+        e1 = e1 || [];
+        e2 = e2 || [];
+        return [...e1, ...e2];
+    });
+}
+
+
+// const volumeNames = new Set<string>();
+
+// export interface Volume extends cloud.Volume {
+//     getVolumeName(): any;
+//     getHostPath(): any;
+// }
+
+// // _Note_: In the current EFS-backed model, a Volume is purely virtual - it
+// // doesn't actually manage any underlying resource.  It is used just to provide
+// // a handle to a folder on the EFS share which can be mounted by container(s).
+// // On platforms like ACI, we may be able to actually provision a unique File
+// // Share per Volume to keep these independently manageable.  For now, on AWS
+// // though, we rely on this File Share having been set up as part of the ECS
+// // Cluster outside of @pulumi/cloud, and assume that that data has a lifetime
+// // longer than any individual deployment.
+// export class SharedVolume extends pulumi.ComponentResource implements Volume, cloud.SharedVolume {
+//     public readonly kind: cloud.VolumeKind;
+//     public readonly name: string;
+
+//     constructor(name: string, opts?: pulumi.ResourceOptions) {
+//         if (volumeNames.has(name)) {
+//             throw new Error("Must provide a unique volume name");
+//         }
+//         super("cloud:volume:Volume", name, {}, opts);
+//         this.kind = "SharedVolume";
+//         this.name = name;
+//         volumeNames.add(name);
+//     }
+
+//     getVolumeName() {
+//         // Ensure this is unique to avoid conflicts both in EFS and in the
+//         // TaskDefinition we pass to ECS.
+//         return utils.sha1hash(`${pulumi.getProject()}:${pulumi.getStack()}:${this.kind}:${this.name}`);
+//     }
+
+//     getHostPath() {
+//         const cluster = getCluster();
+//         if (!cluster || !cluster.efsMountPath) {
+//             throw new Error(
+//                 "Cannot use 'Volume'.  Configured cluster does not support EFS.",
+//             );
+//         }
+//         // Include the unique `getVolumeName` in the EFS host path to ensure this doesn't
+//         // clash with other deployments.
+//         return `${cluster.efsMountPath}/${this.name}_${this.getVolumeName()}`;
+//     }
+// }
+
+// export class HostPathVolume implements cloud.HostPathVolume {
+//     public readonly kind: cloud.VolumeKind;
+//     public readonly path: string;
+
+//     constructor(path: string) {
+//         this.kind = "HostPathVolume";
+//         this.path = path;
+//     }
+
+//     getVolumeName() {
+//         return utils.sha1hash(`${this.kind}:${this.path}`);
+//     }
+
+//     getHostPath() {
+//         return this.path;
+//     }
+// }
```

Comment:
Currently not implemented.  It may not be necessary at all given that users can just supply these values when creating the Service. But i haven't looked into this more to see how this all works.